### ISSUE TITLE
[Bug]: Accessibility issue for Dropdown component when contains separator

### ISSUE THREAD

**Issue by KeiraYang** [microsoft/fluentui#26949]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (110.0.1587.49)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown

## Actual Behavior
fill this out
![image](https://user-images.githubusercontent.com/28148759/220891403-b44ba0a5-87d5-49cc-9d5e-b6707334e029.png)
![image](https://user-images.githubusercontent.com/28148759/220891435-e70c992e-c1ac-4bf8-9a51-89f99790e0ef.png)
![image](https://user-images.githubusercontent.com/28148759/220892192-edc94d71-793e-4dfa-b1a1-b2fbd60eb124.png)

## Expected Behavior
fill this out
For listbox element, the allow children are option element only. If have separator to split the listbox, then either the separator should be remove role, role="none" or separated list to be 2 lists instead of 1.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
